services/frontend/adapters/inbound/controller.go

```diff
@@ -0,0 +1,136 @@
+package inboundAdapters
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"strconv"
+
+	corev1 "k8s.io/api/core/v1"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
+	crlog "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	wbhk "sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"github.com/go-logr/zerologr"
+	"github.com/rs/zerolog/log"
+)
+
+var (
+	healthcheckPort = ":8081"
+	healthcheckPath = "healthz"
+)
+
+type manifestController struct {
+	mgr                        manager.Manager
+	ctx                        context.Context
+	healthcheckClient          *http.Client
+	validationWebhookPort      int
+	validationWebhookCertDir   string
+	validationWebhookPath      string
+	validationWebhookNamespace string
+}
+
+func NewManifestController(ctx context.Context) (*manifestController, error) {
+	// lookup environment variables
+	portString, err := lookupEnv("WEBHOOK_TLS_PORT")
+	if err != nil {
+		return nil, err
+	}
+	port, err := strconv.Atoi(portString)
+	if err != nil {
+		return nil, err
+	}
+	namespace, err := lookupEnv("NAMESPACE")
+	if err != nil {
+		return nil, err
+	}
+	certDir, err := lookupEnv("WEBHOOK_CERT_DIR")
+	if err != nil {
+		return nil, err
+	}
+	webhookPath, err := lookupEnv("WEBHOOK_PATH")
+	if err != nil {
+		return nil, err
+	}
+
+	// create http.Client for healthech requests
+	client := &http.Client{}
+
+	// create ManifestController object
+	var mc = manifestController{
+		ctx:                        ctx,
+		healthcheckClient:          client,
+		validationWebhookPort:      port,
+		validationWebhookNamespace: namespace,
+		validationWebhookPath:      webhookPath,
+		validationWebhookCertDir:   certDir,
+	}
+
+	// Setup the controler manager for input-manifest
+	mc.mgr, err = ctrl.NewManager(config.GetConfigOrDie(), ctrl.Options{
+		Namespace:              mc.validationWebhookNamespace,
+		HealthProbeBindAddress: healthcheckPort,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("unable to set up manifest-controller: " + err.Error())
```

Comment:
`fmt.Errorf("unable to set up manifest-controller: " + err.Error())` -> `fmt.Errorf("unable to set up manifest-controller : %w", err)`